feat(animations): Add compositor thread on Android (#5587)
unoplatform/uno: doc/articles/composition.md

Uno does support few APIs of the [`Windows.UI.Composition` namespace](https://docs.microsoft.com/en-us/uwp/api/windows.ui.composition?view=winrt).

** On Android, composition APIs are functionnal only for Android 10 and upper (API 29).**

## Compositor Thread [Android]

On Android, the composition refers to the [`Draw`](https://developer.android.com/reference/android/view/View#draw(android.graphics.Canvas)) and [`OnDraw`](https://developer.android.com/reference/android/view/View#onDraw(android.graphics.Canvas)) methods.
To get more info about custom drawing on Android, you can refer to the [Android's documentation](https://developer.android.com/training/custom-views/custom-drawing).

By default, those methods are invoked on the UI Thread.

With Uno, you can request to run those methods on a dedicated thread by setting in your Android application's constructor:

```csharp
Uno.CompositionConfiguration.Configuration = Uno.CompositionConfiguration.Options.Enabled;
```

This thread will also be used for [independent animations](https://docs.microsoft.com/en-us/windows/uwp/design/motion/storyboarded-animations#dependent-and-independent-animations).

> **When overriding the `[On]Draw` methods, it's really important to not access any state that can be edited from the UI Thread, including any `DependencyProperty`.**

Reviewer: ```suggestion
**When overriding the `[On]Draw` methods, it is really important not to access any state that can be edited from the UI Thread, including any `DependencyProperty`.**
```

> **Instead, you should capture the state of your control into a `RenderNode` during the `ArrangeOverride` and render it on the provided `Canvas`.**

Reviewer: Should we mention an internal implementation details for android only in the general description ?

Author: As it's in a Android only section, I thought it was easier to put it right here. Maybe @davidjohnoliver would suggest a common pattern in doc for that kind of details?

Reviewer 2: The content aimed at users of Uno should go in one doc, the content aimed at Uno contributors (and anyone curious about internal implementation details) should go in another 

put the second doc here: https://github.com/unoplatform/uno/tree/master/doc/articles/uno-development

and add them both to the TOC ;)
 https://github.com/unoplatform/uno/blob/master/doc/articles/toc.yml

> _There is few known issues associated with the used of the compositor thread, [make sure to read them](#known_issues)._

Reviewer: ```suggestion
_There are few known issues associated with the used of the compositor thread, [make sure to read the section below](#known_issues)._
```

## Knwon issues

> * When using the compositor thread, the native ripple effect of Android (used in native buttons) will not work anymore.

Reviewer: ```suggestion
* When using the compositor thread, the native ripple effect of Android (used in native buttons) does not work.
```
